opm/simulators/linalg/PressureTransferPolicy.hpp

```diff
@@ -74,16 +74,24 @@ class PressureTransferPolicy
     {
         using CoarseMatrix = typename CoarseOperator::matrix_type;
         const auto& fineLevelMatrix = fineOperator.getmat();
-        coarseLevelMatrix_.reset(new CoarseMatrix(fineLevelMatrix.N(), fineLevelMatrix.M(), CoarseMatrix::row_wise));
-        auto createIter = coarseLevelMatrix_->createbegin();
-
+        const std::size_t average_elements_per_row
+            = static_cast<std::size_t>(std::ceil(fineLevelMatrix.nonzeroes() / fineLevelMatrix.N()));
+        const double overflow_fraction = 1.2;
+
+        // construct coarse matrix with implicit compression, to ensure values are continuous in memory
```

Comment:
I was not happy with this change either but this was a fix that worked and then I was planing to come back and investigate it further at a later stage, which I now have. It turned out when constructing the coarse matrix with the `row_wise` build mode, the constructor was missing a number of nonzeros argument, such that it would allocate 0 values in the construction and rather do it when we start setting up the matrix or when we set the first value. This probably made the pointer to the first element change or resulted in a non-contiguous memory layout. I added this `nnz` argument and now it works with build mode `row_wise`.